server/src/main/java/io/crate/user/Role.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.user;
+
+import java.util.Objects;
+import java.util.Set;
+
+import org.jetbrains.annotations.Nullable;
+
+import io.crate.metadata.pgcatalog.OidHash;
+
+public class Role {
+
+    private final String name;
+    private final UserPrivileges privileges;
+    @Nullable
+    private final SecureHash password;
+
+    protected Role(String name, Set<Privilege> privileges, @Nullable SecureHash password) {
+        this.name = name;
+        this.privileges = new UserPrivileges(privileges);
+        this.password = password;
+    }
+
+    public static Role of(String name) {
+        return new Role(name, Set.of(), null);
+    }
+
+    public static Role of(String name, SecureHash password) {
+        return new Role(name, Set.of(), password);
+    }
+
+    public static Role of(String name, @Nullable Set<Privilege> privileges, SecureHash password) {
+        return new Role(name, privileges == null ? Set.of() : privileges, password);
+    }
+
+    public String name() {
+        return name;
+    }
+
+    @Nullable
+    public SecureHash password() {
+        return password;
+    }
+
+    public boolean isUser() {
+        return false;
+    }
+
+    public boolean isSuperUser() {
+        return false;
+    }
```

Comment:
if role cannot be a superuser, can we drop this method here and removed usages in this class?

Reply:
ok, actually need to adjust it if we use only Role class (according to https://github.com/crate/crate/pull/15109#discussion_r1412241946)